### PR TITLE
Bump Cumulus, Polkadot, and Substrate

### LINKED ISSUE #102: Get Canvas running on Rococo again
The parachain is currently no longer working.

### ToDo

- [ ] Update this repository to a newer `cumulus` version
- [ ] Ping devops to upgrade the on-chain runtime

### PR DESCRIPTION
**Pull request by HCastano**
This is ready for review, but I wouldn't merge it just yet. I haven't been able to get it
working locally with `polkadot-launch` due to an issue on the Polkadot side (https://github.com/paritytech/polkadot/issues/4238).

Would partially close #102.

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -526,12 +528,22 @@ match_type! {
...
 
 pub type Barrier = (
 	TakeWeightCredit,
 	AllowTopLevelPaidExecutionFrom<Everything>,
 	AllowUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
 	// ^^^ Parent and its exec plurality get free execution
+	// Expected responses are OK.
+	AllowKnownQueryResponses<PolkadotXcm>,
+	// Subscriptions for version tracking are OK.
+	AllowSubscriptionsFrom<ParentOrSiblings>,
```
Comment (HCastano) [inline runtime/src/lib.rs:546]: @KiChjang would you be able to explain what a `Sibling` is in the context of XCM?

Comment (KiChjang) [reply runtime/src/lib.rs:546]: No surprise here -- siblings are MultiLocations that have the same parent, as denoted by the definition `MultiLocation { parent: 1, junctions: X1(_) }`. It's equivalent to locating for a file under `../*`, where `..` is "up one level" (meaning parent) and "*" is the wildcard operator.

Comment (HCastano) [reply runtime/src/lib.rs:546]: Gotcha, thanks!

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -558,7 +570,7 @@ parameter_types! {
 }
 
 /// No local origins on this chain are allowed to dispatch XCM sends/executions.
-pub type LocalOriginToLocation = SignedToAccountId32<Origin, AccountId, RelayNetwork>;
+pub type LocalOriginToLocation = ();
```
Comment (cmichi) [inline runtime/src/lib.rs:573]: I guess this is something that we'll need to change for `canvas`.

**Review by cmichi — APPROVED** [review]

File: runtime/src/lib.rs (hunk 2 above)
Comment (HCastano) [reply runtime/src/lib.rs:573]: Yeah, we'll need to definitely revisit our XCM settings after we a) learn more about what they actually do and b) determine how we actually want to use XCM in the context of Canvas

**Comment by HCastano** [comment]
@cmichi there were some small changes with the new version bump, can you take another quick look?

**Comment by HCastano** [comment]
I've tested this with `polkadot-launch` - I can deploy and interact with an ERC-20 contract, so I think we're good

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -23,6 +23,11 @@ use serde::{Deserialize, Serialize};
 use sp_core::{crypto::UncheckedInto, sr25519, Pair, Public};
 use sp_runtime::traits::{IdentifyAccount, Verify};
 
+/// The ID of this Parachain as registered on the Relay Chain.
+///
+/// TODO: When we register as a common good parachain this will need to change.
+pub const PARA_ID: u32 = 12648430;
```
Comment (cmichi) [inline node/src/chain_spec.rs:29]: I think there is already a parachain id which @lovelaced assigned for the Rococo deployment?

Comment (lovelaced) [reply node/src/chain_spec.rs:29]: Canvas on Rococo is 1002 - and as far as I understand common good chains should have a para ID 1000 <= x < 2000.

**Review by cmichi — APPROVED** [review]

**Pull request merged**